### ISSUE TITLE
Code organization & conventions

### ISSUE THREAD

**Issue by PeterPablo** [matlab2tikz/matlab2tikz#445]
Since your simplification efforts comes with many new function names, would it make sense to discuss the naming? For me, as someone new to the actual code, it is easier to read/interpret something like ColorDefinitions_generate, if there are several functions of the same 'scope', like e.g. with opts_*.

**Egon Geerardyn** replied in https://github.com/nschloe/matlab2tikz/commit/4fa485968b0912b59eb98402112ce5165a9a0c5e#commitcomment-8528009:

> I agree with you that there is limited consistency in the names I chose (especially the opts_… ones are odd compared to all others).
> 
> Let me explain my logic there: the option arrays (i.e. anything related to opts_…) used very different functions before: addToOptions, prettyPrintOpts, merge and I always forgot how to write those exactly. 
> This opts is a first step of mine towards an OOP design of that part of the code, where it is more common to have naming like "object.method" (or "object_method" in this case). One day, I hope to replace them with what I'd call a faux-objects (struct with function handles) such that the code will likely become options = options.add(key, value) instead of the current intermediate step options = opts_add(options, key, value).
> 
> To be honest, at the moment I'm not completely convinced that the current opts_... is good style. When you consider opts an isolated module of the code, I think I made an improvement. But consistency-wise for the whole of m2t, it may be debatable.
> 
> Most other functions, however, behave in a more procedural fashion and there I prefer names that read more like prose (i.e. verbName instead of nameVerb). That's also my reason for those other names (and 99% of the code).
> 
> TL;DR: Some parts are more procedural, there I used "verbNoun", other parts are more OOP/modular and in those parts I use "noun_verb" (as intermediate step to faux-OOP).
> 
> But if you wish to discuss things further, feel free to open an issue with the "Discussion" label to discuss about code standards, naming conventions, ... That is easier to keep track of than comments to commits.

**Comment by okomarov** [comment]
#### Naming convention for `get/set` methods.

There is a bit of **confusion** in the naming of functions and the use of `get*` and `set*` prefixes for (a subset of) `m2t` subfunctions should be a natural choice. For many, we are de facto just getting some Matlab/Octave properties and translating them to `tikz` and hence setting *equivalent* pgfplots properties. 

I propose to sort by **graphical objects**, e.g. Axis, and then by **property** and use the **singular** for both (grammar should not play a role here):

`get/setObjectnameProperty()`

So, all of the Axes methods will become:
`getAxisPosition` (instead of `getAxesPosition`)
`getAxisRelativePosition` (instead of `getRelativeAxesPosition`)

It is much easier to see the methods belonging to one graphical object and eventually even spot potential repetitions among graphical objects. Now, you cannot spot the `getRelativeAxesPosition` method below, but you would if it were renamed:

![untitled](https://cloud.githubusercontent.com/assets/3731173/7101671/f8e35dee-e05b-11e4-94b0-92846eac425e.png)

If a method is equal/shared among graphical objects, then it simply becomes:
`get/setProperty()`

Now, I mentioned that some methods are a bit confusing, since they do not do what they imply or do too many things. A couple of examples:
* `getTitleOrLabel_()` First gets and then also sets. I was reading through the code and wondering when some label is set...couldn;t find it and looked through the code and realized what was going on.
* `retrievePositionOfAxes()` This is a good one. I thought, why don't we just call it `getAxisPosition()` instead of the longer `retrieve...`? Because, it already contains a call to `getAxesPosition`, and it is actually a (conditional) setter method :)

I think it would also be easier to get started as a developer (#566) if the naming was a bit clearer and organized.

**Comment by egeerardyn** [comment]
I agree with the ordering of the names, but I don't with the example that you want to rename Axes to Axis. The object class is Axes (at least in HG1, I'm unsure about HG2), to indicate the whole (xy or xyz) axes cross , not a single (x, y or z) axis. We should keep the same nomenclature as Matlab IMHO.

**Comment by okomarov** [comment]
Right! Matlab nomenclature first and singular case if uncertain (?).

**Comment by egeerardyn** [comment]
Agree.

**Comment by PeterPablo** [comment]
:+1:
